### PR TITLE
Build with expected versions

### PR DESCRIPTION
**Pull request by jjcarstens**
* Run the formatter for new elixir versions
* update xterm 4.18.0
* Also fixes the case where a known CA is expired and will prevent device connection
  * An integration test was also added to ensure devices with an expired cert can still connect if the device cert was previously registered

### CODE REVIEW

File: apps/nerves_hub_device/test/nerves_hub_device_web/channels/websocket_test.exs [hunk 1]
```diff
@@ -284,7 +284,7 @@ defmodule NervesHubDeviceWeb.WebsocketTest do
       {device, _firmware} = device_fixture(user, %{identifier: @valid_serial}, org)
 
       not_before = Timex.now() |> Timex.shift(days: -1)
-      not_after = Timex.now() |> Timex.shift(seconds: 1)
+      not_after = Timex.now() |> Timex.shift(days: 1)
```
Comment (tonnenpinguin) [inline apps/nerves_hub_device/test/nerves_hub_device_web/channels/websocket_test.exs:287]: I think that effectively makes the test obsolete as the signer isn't actually expired when the request comes in

Comment (jjcarstens) [reply apps/nerves_hub_device/test/nerves_hub_device_web/channels/websocket_test.exs:287]: O, duh 🤦 i totally mixed this up as being a different test. I’ll have to play with this a bit more

**Review by tonnenpinguin — APPROVED** [review]

File: apps/nerves_hub_device/lib/nerves_hub_device/ssl.ex [hunk 2]
```diff
@@ -60,6 +60,20 @@ defmodule NervesHubDevice.SSL do
...
+    # to register, the validation will fail later on due to the expired.
+    #
+    # If this is an existing device presenting an expired cert in the chain
+    # then allowing the expired CA cert prevents the request from being
+    # terminating prematurely 
+    case check_known_ca(otp_cert) do
+      {:ok, _ca} -> {:valid, state}
+      _unknown_ca -> :cert_expired
+    end
+  end
```
[the remark this answers is not in the document]
Comment (jjcarstens) [reply apps/nerves_hub_device/lib/nerves_hub_device/ssl.ex]: Fixed

File: apps/nerves_hub_device/test/nerves_hub_device_web/channels/websocket_test.exs [hunk 3]
```diff
@@ -279,6 +279,65 @@ defmodule NervesHubDeviceWeb.WebsocketTest do
       refute_receive({"presence_diff", _})
     end
 
+    test "already registered expired certificate can connect", %{user: user} do
```
[the remark this answers is not in the document]
Comment (tonnenpinguin) [reply apps/nerves_hub_device/test/nerves_hub_device_web/channels/websocket_test.exs:282]: Ahhh yeah, sorry about that :)

[the remark this answers is not in the document]
Comment (jjcarstens) [reply apps/nerves_hub_device/test/nerves_hub_device_web/channels/websocket_test.exs:282]: Missed the timer in the other test. I updated it to match this new one and definitely force the cert to be expired

**Pull request merged**